tools/input-segments/src/InputSegments.ts

```diff
@@ -257,129 +261,218 @@ export class InputSegments extends TextfieldBase {
         };
 
         /**
-         * TODO: Include ARIA attributes for editable segments
-         * TODO: Use `@input`/`@beforeinput` events to handle data input/content cleanup
-         * TODO: Move `handleKeydown()` call to a cache so that it doesn't cycle on the binding in each render pass
```

Comment:
This doesn't seem to be addressed here. The issue that is looks to resolve seems to exist in the `beforeinput` and `input` event listeners as well, now.

Reply:
In fact, this `TODO` was not resolved, it only had the text adjusted, as previously we only had `keydown`, but now we also have two new events, `input` and `beforeinput`.

Even so, we will need help with this improvement, because although we have talked about this in the past, I don't remember how we could actually improve this process, using what you called "call to a cache".